plugins/field-colour/src/blocks/colourBlend.ts

```diff
@@ -0,0 +1,274 @@
+/**
+ * @license
+ * Copyright 2024 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Block, common as BlocklyCommon} from 'blockly';
+import { JavascriptGenerator, Order as JavascriptOrder } from 'blockly/javascript';
+import { DartGenerator, Order as DartOrder } from 'blockly/dart';
+import { LuaGenerator, Order as LuaOrder } from 'blockly/lua';
+import { PhpGenerator, Order as PhpOrder } from 'blockly/php';
+import { PythonGenerator, Order as PythonOrder } from 'blockly/python';
+import { registerFieldColour } from '../field_colour';
+import { Generators } from './generatorUtils';
+
+
+const BLOCK_NAME = 'colour_blend';
+
+// Block for blending two colours together.
+const jsonDef =
+{
+    'type': BLOCK_NAME,
+    'message0':
+        '%{BKY_COLOUR_BLEND_TITLE} %{BKY_COLOUR_BLEND_COLOUR1} ' +
+        '%1 %{BKY_COLOUR_BLEND_COLOUR2} %2 %{BKY_COLOUR_BLEND_RATIO} %3',
+    'args0': [
+        {
+            'type': 'input_value',
+            'name': 'COLOUR1',
+            'check': 'Colour',
+            'align': 'RIGHT',
+        },
+        {
+            'type': 'input_value',
+            'name': 'COLOUR2',
+            'check': 'Colour',
+            'align': 'RIGHT',
+        },
+        {
+            'type': 'input_value',
+            'name': 'RATIO',
+            'check': 'Number',
+            'align': 'RIGHT',
+        },
+    ],
+    'output': 'Colour',
+    'helpUrl': '%{BKY_COLOUR_BLEND_HELPURL}',
+    'style': 'colour_blocks',
+    'tooltip': '%{BKY_COLOUR_BLEND_TOOLTIP}',
+};
+
+/**
```

Comment:
Open question: How should I describe these functions? I want to settle on a general-purpose description that I can copy into every plugin, or decide on a shorthand with reference to some other documentation.

Reply:
I'd suggest referencing https://developers.google.com/blockly/guides/create-custom-blocks/code-generation/overview#block-code_generators .  Beyond that… 🤔

I think "block code generators" or "block generator functions" are both OK.  The former we already use in some of our docs; the latter is I think slightly more clear (they are just a function, and not a kind of `CodeGenerator`).